### PR TITLE
Further refactor of chatdown and fix 1 bug

### PR DESCRIPTION
**Pull request by jspruance**
- Write to output dir (--out) was only working when a dir passed in (--in) but not for single files. This is fixed.

### CODE REVIEW

File: packages/chatdown/src/commands/chatdown.ts [hunk 1]
```diff
@@ -37,12 +37,14 @@ export default class Chatdown extends Command {
         })
       }
 
+      let outputDir = (flags.out) ? flags.out.trim() : './'
+      if (outputDir.substr(0, 2) === './') {
+        outputDir = path.resolve(process.cwd(), outputDir.substr(2))
```
Comment (munozemilio) [inline packages/chatdown/src/commands/chatdown.ts]: let outputDir = path.resolve(flags.out) should return the same result

File: packages/chatdown/src/commands/chatdown.ts [hunk 2]
```diff
@@ -91,21 +94,27 @@ export default class Chatdown extends Command {
...
+  private getFileName(file: any) {
+    let fileName = file
+    if (file.lastIndexOf('/') !== -1) {
+      fileName = fileName.substr(fileName.lastIndexOf('/') + 1)
+    }
+    if (file.lastIndexOf('\\') !== -1) {
+      fileName = fileName.substr(fileName.lastIndexOf('\\') + 1)
+    }
+    fileName = fileName.substring(0, fileName.lastIndexOf('.'))
+    return fileName
```
Comment (munozemilio) [inline packages/chatdown/src/commands/chatdown.ts]: let fileName = path.basename(file, path.extname(file)) probably does the same

File: packages/chatdown/src/commands/chatdown.ts [hunk 3]
```diff
@@ -119,7 +128,13 @@ export default class Chatdown extends Command {
     })
   }
 
-  private async writeOut(activities: any) {
+  private async writeOut(activities: any, fileName: string, outputDir: any) {
+    if (fileName && outputDir) {
+      let writeFile = `${outputDir}\\${fileName}.transcript`
```
Comment (munozemilio) [inline packages/chatdown/src/commands/chatdown.ts]: let writeFile = path.join(outputDir, ${fileName}.transcript)

File: packages/chatdown/src/commands/chatdown.ts (hunk 1 above)
Comment (jspruance) [reply packages/chatdown/src/commands/chatdown.ts]: Updated old lib code to use path library

File: packages/chatdown/src/commands/chatdown.ts (hunk 2 above)
Comment (jspruance) [reply packages/chatdown/src/commands/chatdown.ts]: updated

File: packages/chatdown/src/commands/chatdown.ts (hunk 3 above)
Comment (jspruance) [reply packages/chatdown/src/commands/chatdown.ts]: updated

File: packages/chatdown/src/commands/chatdown.ts [hunk 4]
```diff
@@ -37,12 +37,15 @@ export default class Chatdown extends Command {
         })
       }
 
+      // let outputDir = (flags.out) ? flags.out.trim() : './'
```
Comment (munozemilio) [inline packages/chatdown/src/commands/chatdown.ts]: Can we remove comments?

File: packages/chatdown/src/commands/chatdown.ts [hunk 5]
```diff
@@ -37,12 +37,15 @@ export default class Chatdown extends Command {
...
+      // if (outputDir.substr(0, 2) === './') {
+      //   outputDir = path.resolve(process.cwd(), outputDir.substr(2))
+      // }
+      let outputDir = flags.out ? path.resolve(flags.out) : path.resolve(process.cwd())
```
Comment (munozemilio) [inline packages/chatdown/src/commands/chatdown.ts]: if flags.out is empty should we print to console?

File: packages/chatdown/src/commands/chatdown.ts (hunk 4 above)
Comment (jspruance) [reply packages/chatdown/src/commands/chatdown.ts]: oops, removed!

File: packages/chatdown/src/commands/chatdown.ts (hunk 5 above)
Comment (jspruance) [reply packages/chatdown/src/commands/chatdown.ts]: i don't think so, since 'out' is totally optional

**Pull request merged**